components/raftstore-v2/src/operation/command/admin/merge/commit.rs

```diff
@@ -0,0 +1,753 @@
+        let mut request = new_admin_request(sibling_region.get_id(), sibling_peer.clone());
+        request
+            .mut_header()
+            .set_region_epoch(sibling_region.get_region_epoch().clone());
+        let mut admin = AdminRequest::default();
+        admin.set_cmd_type(AdminCmdType::CommitMerge);
+        admin.mut_commit_merge().set_source(self.region().clone());
+        admin.mut_commit_merge().set_commit(state.get_commit());
+        admin.mut_commit_merge().set_entries(entries.into());
+        request.set_admin_request(admin);
+        // Please note that, here assumes that the unit of network isolation is store
+        // rather than peer. So a quorum stores of source region should also be the
+        // quorum stores of target region. Otherwise we need to enable proposal
+        // forwarding.
+        let (msg, _) = PeerMsg::admin_command(request);
+        store_ctx
+            .router
+            .force_send(target_id, msg)
+            .map_err(|_| Error::RegionNotFound(target_id))
+    }
+
+    // Match v1::validate_merge_peer.
+    fn validate_merge_peer<T>(
+        &self,
+        store_ctx: &mut StoreContext<EK, ER, T>,
+        target_region: &metapb::Region,
+    ) -> Result<bool> {
+        let target_region_id = target_region.get_id();
+        let exist_region = {
+            let store_meta = store_ctx.store_meta.lock().unwrap();
+            store_meta
+                .regions
+                .get(&target_region_id)
+                .map(|(r, _)| r.clone())
+        };
+        if let Some(r) = exist_region {
+            let exist_epoch = r.get_region_epoch();
+            let expect_epoch = target_region.get_region_epoch();
+            // exist_epoch > expect_epoch
+            if util::is_epoch_stale(expect_epoch, exist_epoch) {
+                return Err(box_err!(
+                    "target region changed {:?} -> {:?}",
+                    target_region,
+                    r
+                ));
+            }
+            // exist_epoch < expect_epoch
+            if util::is_epoch_stale(exist_epoch, expect_epoch) {
+                info!(
+                    self.logger,
+                    "target region still not catch up, skip.";
+                    "target_region" => ?target_region,
+                    "exist_region" => ?r,
+                );
+                return Ok(false);
+            }
+            return Ok(true);
+        }
+
+        let mut msg = Box::<RaftMessage>::default();
+        msg.set_region_id(target_region.get_id());
+        msg.set_from_peer(self.peer().clone());
+        let local_target_peer = find_peer(target_region, store_ctx.store_id)
+            .cloned()
+            .unwrap();
+        msg.set_to_peer(local_target_peer);
+        msg.set_region_epoch(target_region.get_region_epoch().clone());
+        let extra_msg = msg.mut_extra_msg();
+        extra_msg.set_type(ExtraMessageType::MsgAvailabilityRequest);
```

Comment:
What's the purpose?

Reply:
I thought you meant something like this in https://github.com/tikv/tikv/pull/14253#discussion_r1115274309? If response for this is unavailable (`availability_report` is unset), the merge will be rolled back.

In addition, I was about to use this type of request to check trimmed status too. The response will carry a `trimmed` flag.